### PR TITLE
feat: adding relevant files for block list

### PR DESCRIPTION
**Pull request by Nilay27**
* Adding relevant files for block list contract.
* Currently starting out with the assumption that it does not need to be upgradeable, as I do not see any need for it. Let me know if otherwise.

### CODE REVIEW

File: apps/smart-contracts/token/contracts/BlockList.sol [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity =0.8.7;
+
+import "./interfaces/IBlockList.sol";
+import "prepo-shared-contracts/contracts/SafeOwnable.sol";
+
+contract BlockList is IBlockList, SafeOwnable {
+  constructor(address _nominatedOwner) {}
+
+  function set(address[] calldata accounts, bool[] calldata blocked) external override onlyOwner {}
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/contracts/BlockList.sol:10]:
```suggestion
  function set(address[] calldata _accounts, bool[] calldata _blocked) external override onlyOwner {}
```

File: apps/smart-contracts/token/contracts/BlockList.sol [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+  function reset(address[] calldata blockedAccounts) external override onlyOwner {}
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/contracts/BlockList.sol:12]:
```suggestion
  function reset(address[] calldata _blockedAccounts) external override onlyOwner {}
```

**Review by ramenforbreakfast — APPROVED** [review]
LGTM besides some nits.

File: apps/smart-contracts/token/contracts/interfaces/IBlockList.sol [hunk 3]
```diff
@@ -2,7 +2,7 @@
 pragma solidity =0.8.7;
 
 //TODO: add all natspecs at the end
-interface IBlocklist {
+interface IBlockList {
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/interfaces/IBlockList.sol:5]: nit:
```suggestion
interface IBlocklist {
```
'blocklist' and 'allowlist' are single words

File: apps/smart-contracts/token/contracts/BlockList.sol [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity =0.8.7;
+
+import "./interfaces/IBlockList.sol";
+import "prepo-shared-contracts/contracts/SafeOwnable.sol";
+
+contract BlockList is IBlockList, SafeOwnable {
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/BlockList.sol:7]: nit:
```suggestion
contract Blocklist is IBlocklist, SafeOwnable {
```
and same for file name and test file name

File: apps/smart-contracts/token/test/BlockList.test.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { expect } from 'chai'
+import { blockListFixture } from './fixtures/BlockListFixtures'
+import { BlockList } from '../types/generated'
+
+describe('=> BlockList', () => {})
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/BlockList.test.ts:5]:
```suggestion
describe('=> Blocklist', () => {})
```

File: apps/smart-contracts/token/test/fixtures/BlockListFixtures.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { ethers } from 'hardhat'
+import { BlockList } from '../../types/generated'
+
+export async function blockListFixture(nominatedOwnerAddress: string): Promise<BlockList> {
+  const Factory = await ethers.getContractFactory('BlockList')
+  return (await Factory.deploy(nominatedOwnerAddress)) as unknown as BlockList
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/fixtures/BlockListFixtures.ts:6]:
```suggestion
export async function blocklistFixture(nominatedOwnerAddress: string): Promise<Blocklist> {
  const Factory = await ethers.getContractFactory('Blocklist')
  return (await Factory.deploy(nominatedOwnerAddress)) as unknown as Blocklist
```
and file name too

**Review by XavierEkkel — APPROVED** [review]
LGTM after naming nit

**Pull request merged**